webui/react/src/pages/TaskList.tsx

```diff
@@ -108,6 +113,66 @@ const TaskList: React.FC = () => {
     triggerTensorboardsRequest,
   ]);
 
+  const handleSourceExpand = useCallback((event: React.MouseEvent, id: string) => {
+    event.stopPropagation();
+    setSourceExpanded(prev => {
+      const newSourceExpanded = { ...prev };
+      if (!newSourceExpanded[id]) newSourceExpanded[id] = true;
+      else delete newSourceExpanded[id];
+      return newSourceExpanded;
+    });
+  }, []);
+
+  const columns = useMemo(() => {
+    const sourceRenderer: TaskRenderer = (_, record) => {
+      const info = {
+        isPlural: false,
+        label: '',
+        path: '',
+        source: [] as number[],
+      };
+      if (record.misc?.experimentIds) {
+        info.label = 'Experiment';
+        info.path = '/ui/experiments';
+        info.source = record.misc.experimentIds || [];
+      } else if (record.misc?.trialIds) {
+        info.label = 'Trial';
+        info.path = '/ui/trials';
+        info.source = record.misc.trialIds || [];
+      }
+      info.isPlural = info.source.length > 1;
+      info.source.sort(numericSorter);
+
+      const isExpanded = sourceExpanded[record.id];
+      const sourceCount = info.source.length;
+      const showToggle = info.source.length > MAX_SOURCES;
+      const toggleLabel = `${isExpanded ? 'Hide' : 'Show'} ${sourceCount}${isExpanded ? '' : '+'}`;
+
+      return sourceCount !== 0 ? (
+        <div className={css.sourceLinks}>
+          <label>{info.label}{info.isPlural ? 's' : ''}</label>
+          {info.source.map((id, index) => {
+            const display = index < MAX_SOURCES || isExpanded ? 'inline' : 'none';
+            const handleClick = (event: React.MouseEvent) => {
+              event.stopPropagation();
+              makeClickHandler(`${info.path}/${id}`)(event);
+            };
+            return <a key={id} style={{ display }} onClick={handleClick}>{id}</a>;
```

Comment:
Non-blocking nitpick. When I click 'show' or 'hide', it dynamically resizes the columns and changes the position of all of the columns slight, which I find annoying. Not a big deal, but if you have an easy solution, I think it would nice to not have the UI change quite as much when I click `Show`.

<img width="1624" alt="Screen Shot 2020-08-10 at 11 28 32 AM" src="https://user-images.githubusercontent.com/13806811/89817262-b6b2b400-dafc-11ea-8134-e8f057188335.png">
<img width="1631" alt="Screen Shot 2020-08-10 at 11 28 38 AM" src="https://user-images.githubusercontent.com/13806811/89817270-b9150e00-dafc-11ea-9884-b656888b00fc.png">

Reply:
The only way I know around this is to have fixed sized columns, but early on when we were using them, it created some weird rendering for linux browsers (had a gap on the far right hand said of the header columns). I'm hoping to move away from the column based sources relatively soon, post migration so this should not be an issue after the refactor.